Bundle/Resources/config/psr_factories.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\UploadedFileFactoryInterface;
+use Psr\Http\Message\UriFactoryInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\PsrHttpFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('psr_http_message.psr_http_factory', PsrHttpFactory::class)
+        ->args([
+            new Reference('psr_http_message.psr.server_request_factory'),
+            new Reference('psr_http_message.psr.stream_factory'),
+            new Reference('psr_http_message.psr.uploaded_file_factory'),
+            new Reference('psr_http_message.psr.response_factory'),
+        ])
+
+        ->alias(HttpMessageFactoryInterface::class, 'psr_http_message.psr_http_factory')
+
+        ->alias(RequestFactoryInterface::class, 'psr_http_message.psr.request_factory')
+        ->alias(ResponseFactoryInterface::class, 'psr_http_message.psr.response_factory')
+        ->alias(ServerRequestFactoryInterface::class, 'psr_http_message.psr.server_request_factory')
+        ->alias(StreamFactoryInterface::class, 'psr_http_message.psr.stream_factory')
+        ->alias(UploadedFileFactoryInterface::class, 'psr_http_message.psr.uploaded_file_factory')
+        ->alias(UriFactoryInterface::class, 'psr_http_message.psr.uri_factory')
```

Comment:
there is a soft conflict between these and https://github.com/symfony/recipes/blob/master/nyholm/psr7/1.0/config/packages/nyholm_psr7.yaml

should we remove the recipe for nyholm/psr7 in favor of this bundle?

Reply:
I think we should.

Reply:
I was about to send a PR, but then: ppl won't be able to add only one package to have working psr17 factories. If they install the bridge, they won't have the implementation. And if we they install the implementation and we remove the recipe, they won't have the aliases :(

Reply:
Can you see https://github.com/derrabus/psr-http-message-bridge/pull/1? It's an attempt to support any PSR-17 implementation.

Reply:
Then again, if we just leave everything as is, not much will break. The Nyholm recipe will steal the aliases, but they point to the same stateless classes anyway.

What we could do:
* We could comment out the aliases in the Nyholm recipe and add a note that the developer should either un-comment them or install the bundle for a better integration.
* Add a pack that depends on the bridge and the Nyholm implementation and remove the Nyholm recipe.

Reply:
@ajgarlag Thank you! Let's merge this PR first and move your PR to this repo afterwards.

Reply:
It'd like to give the pack approach a try before merging, because I'm not sure this bundle is the best approach, especially since we're still talking about a pack here...

Reply:
Okay, go ahead.

Reply:
@derrabus I could split my PR into two commits:

1. Conditionally add PSR-17 aliases in a Compiler pass if not defined previously.
2. Support any PSR-17 implementation.

I think that the first one could be included here. WDYT?

Reply:
Let's wait for Nicolas first.